### PR TITLE
Fix broken requirements_boyd_3.9

### PR DESCRIPTION
**Pull request by can-anyscale**
This file is broken https://buildkite.com/ray-project/release/builds/24678#0192d43e-715d-4382-99b3-039ebbbd5c13/6-15800

Broken by https://github.com/ray-project/ray/pull/45328/files

Attempt to fix by this PR; I'll revert the original PR if this doesn't work

Test:
- CI

### CODE REVIEW

**Review by aslonnie — APPROVED** [review]
what is `gym`?

**Comment by can-anyscale** [comment]
it's a API rllib library, replaced by gymnasium in the newer versions (https://pypi.org/project/gymnasium/). Not sure if the test application code needs to be updated too so I'm running all nightly release tests to see if this works, otherwise I'll just revert the blamed PR

**Review by sven1977 — APPROVED** [review]
LGTM.

File: release/ray_release/byod/requirements_byod_3.9.txt [hunk 1]
```diff
@@ -116,30 +115,35 @@ aiosignal==1.3.1 \
...
-    --hash=sha256:ae2ba24557e0ce541ea3be13b148db2a9cfa730d83537b4cbed5e10449826e51 \
-    --hash=sha256:b00f74e27815131c1a2791f3d48114363fa2708e19f09ce6b7b614cb14c9d469 \
-    --hash=sha256:b2aa2f69a4169742800615970efe6914fa856e33eaf7fa9133c0e06a617a80e2 \
-    --hash=sha256:c9b168eb88c87d0f3e2a778e6c5cdde4ad951d1ca8a6dc3d3679fd45398df7d1 \
-    --hash=sha256:d49b550a2d9c25b63c343aa680fd81f253a3714cdc0e1835640933ebff1798ff \
-    --hash=sha256:eadf9f3990b4ff2f9e5ca35889f5e2e95cddd6a353d9d857d9b4601a6e1c4e7c \
-    --hash=sha256:f10b1df8774bbe3b00365748b5e0e07cf35f6a703bbaff991bc7b3b2247dccc9 \
-    --hash=sha256:f278036f9b6066062abcdf0987a0ec5a8e0f22a2c7cfac925e39378d4343d490
+ale-py==0.10.1 \
+    --hash=sha256:076a44a61c2518b844f765692a91d0a6b383c6592b5fdabd94fd24d4c62a54ef \
```
Comment (sven1977) [inline release/ray_release/byod/requirements_byod_3.9.txt:119]: Ah, makes sense. Sorry, I wasn't even thinking about updating these hashes here as well.

**Comment by can-anyscale** [comment]
some release tests are broken https://buildkite.com/ray-project/release/builds/24687#_ unfortunately

**Pull request merged**